File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    var arr=[];
    for(var rt=0; rt<nums.length; rt++){
        if(arr[nums[rt]] && rt+1-arr[nums[rt]]<=k)
            return true;
        arr[nums[rt]]=rt+1;
    }
    return false;
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> m;
        for(uint_fast8_t i=0; i<(uint_fast8_t)nums.size(); i++){
            if(m[nums[i]] && i-m[nums[i]]<k)
                return true;
            m[nums[i]]=i+1;
        }
        return false;
    }
};